django_rest_generator/mixins.py

```diff
@@ -163,7 +167,12 @@ def get(
     ) -> APIResponse:
         url = cls.class_url()
         logger.debug(f"[get] Making GET request to {url} with parameters {params}")
-        return cls.make_request("GET", url=url, params=params)
+        obj = cls.make_request("GET", url=url, params=params)
+        if len(obj.results) > 1:
+            raise ValueError("Got more than one object back from request.")
+        elif len(obj.results) < 1:
+            return None
+        return obj.results[0]
```

Comment:
If we get to merge this change, we'd have to find all uses of this function on dependent code to change the return data handling, besides that, I agree it's a better way.